crates/wasmparser/src/validator/operators.rs

```diff
@@ -959,15 +979,24 @@ impl<'resources, R: WasmModuleResources> OperatorValidatorTemp<'_, 'resources, R
     fn params(&self, ty: BlockType) -> Result<impl PreciseIterator<Item = ValType> + 'resources> {
         Ok(match ty {
             BlockType::Empty | BlockType::Type(_) => Either::B(None.into_iter()),
-            BlockType::FuncType(t) => Either::A(self.func_type_at(t)?.inputs()),
+            BlockType::FuncType(t) => Either::A(self.func_type_at(t)?.inputs().map(|mut ty| {
+                self.resources.canonicalize_valtype(&mut ty);
```

Comment:
Is this necessary? I would have expected that `func_type_at` is looking something up which has already been canonicalized